### PR TITLE
Feat/product list

### PR DESCRIPTION
**Pull request by MargaritaSukhova**
## Describe your changes

- Add product list page for Vendor
- Updated routes with product list page
- Add vendor product table, product table card components
- Moved styled backdrop to shared components
- Add modal popup and action buttons components for product list page
- Add formatToTwoDecimalPlaces util to format price 
- Add test for formatToTwoDecimalPlaces util 

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](#https://ticket-url)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
Снимок экрана 2024-07-14 в 22 52 34
Снимок экрана 2024-07-14 в 22 52 41

### CODE REVIEW

File: src/components/ProductTableCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            alt={name || 'Product image'}
+          />
+        ) : (
+          <Image
+            width="52px"
+            height="67px"
+            src="src/assets/photos/no-image.jpg"
+            alt="Default image"
+          />
+        )}
```
Comment (lyudad) [inline src/components/ProductTableCard/index.tsx]: the difference in 2 images only on url, or condition to url not to full component

File: src/pages/vendor/VendorProductsPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import VendorSectionTitle from '../VendorSectionTitle';
+
+import ProductsTable from './ProductsTable';
+import products from './ProductsTable/productsMock';
+import SectionWrapper from './styles';
+
+function VendorProductPage() {
+  const { t } = useTranslation();
+  const [, setOrder] = useState<SortOrder>(sortOptions.DESC);
```
Comment (lyudad) [inline src/pages/vendor/VendorProductsPage/index.tsx:19]: why state needed if value is not used

File: src/components/ProductTableCard/index.tsx (hunk 1 above)
Comment (MargaritaSukhova) [reply src/components/ProductTableCard/index.tsx]: thank you, fixed it

File: src/pages/vendor/VendorProductsPage/index.tsx (hunk 2 above)
Comment (MargaritaSukhova) [reply src/pages/vendor/VendorProductsPage/index.tsx:19]: I will use order later when I'll be connecting to backend and send order in request body, and backend will sort products in needed order

**Review by lyudad — APPROVED** [review]

**Pull request merged**